src/solidity-frontend/solidity_convert_literals.cpp

```diff
@@ -84,6 +84,39 @@ bool solidity_convertert::convert_string_literal(
 
   return false;
 }
+
+/**
+ * convert hex-string to uint constant
+ * @n: the bit width, default 0 (meaning config.ansi_c.int_width)
+*/
+bool solidity_convertert::convert_hex_literal(
```

Comment:
As there have been some cases where we were wrongly using uint_type() instead of size_type() in the past, I would like to know whether the default value 0 here should really correspond to uint_type(). If, for instance, these hex-literals are used to specify sizes of objects in memory, they should probably default to size_type().

Reply:
Changed as suggested

```c++
  if(n)
    type = unsignedbv_typet(n);
  else  // n = 0 default
    type = size_type();
```